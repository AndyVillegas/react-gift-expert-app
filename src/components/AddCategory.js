import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e)=>{
        setInputValue(e.target.value);
    }
    const handleSubmit = (e)=>{
        e.preventDefault();
        if (valueIsValid()) {
            setCategories(categories=>[ inputValue,...categories]);
            clearInput();
        }
    }
    const clearInput = ()=>{
        setInputValue('');
    }
    const valueIsValid = ()=>inputValue.trim().length>2;
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}