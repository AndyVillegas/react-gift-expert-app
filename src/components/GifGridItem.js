import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {
    return (
        <div className="card">
            <img src={url} alt={title?.length>0?title:'h'}/>
            <div>{title?.length>0?title:'No title'}</div>
        </div>
    )
}
GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}