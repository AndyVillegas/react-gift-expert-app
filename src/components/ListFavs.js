import React from 'react'

export const ListFavs = () => {
    return (
        <div>
            
        </div>
    )
}
