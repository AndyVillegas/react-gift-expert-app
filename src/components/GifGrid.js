import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
        <h3>{category}</h3>
        {loading && <div className="loading">Cargando...</div>}
        <div className="card-container animate__animated animate__fadeInLeft">
            {
                images.map(image => (
                    <GifGridItem
                        key={image.id}
                        {...image}
                    />
                ))
            }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}