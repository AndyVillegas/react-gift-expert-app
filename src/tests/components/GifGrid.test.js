import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import {useFetchGifs} from '../../hooks/useFetchGifs'; 
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en gifgrid', () => {
    const category = 'Naruto';
    test('debe de hacer match con el snapshot', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de mostraar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url:'https://kk.com',
            title: 'Titulo',
        }];
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
})
