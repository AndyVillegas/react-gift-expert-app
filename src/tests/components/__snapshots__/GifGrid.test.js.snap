// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en gifgrid debe de hacer match con el snapshot 1`] = `
<Fragment>
  <h3>
    Naruto
  </h3>
  <div
    className="loading"
  >
    Cargando...
  </div>
  <div
    className="card-container animate__animated animate__fadeInLeft"
  />
</Fragment>
`;

exports[`Pruebas en gifgrid debe de mostraar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3>
    Naruto
  </h3>
  <div
    className="loading"
  >
    Cargando...
  </div>
  <div
    className="card-container animate__animated animate__fadeInLeft"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Titulo"
      url="https://kk.com"
    />
  </div>
</Fragment>
`;
